dds/DCPS/ContentFilteredTopicImpl.cpp

```diff
@@ -59,6 +66,21 @@ ContentFilteredTopicImpl::set_expression_parameters(const DDS::StringSeq& p)
     DDS::RETCODE_OUT_OF_RESOURCES);
 
   const CORBA::ULong len = p.length();
+
+  // Check sequence of strings that give values to the ‘parameters’ (i.e., "%n" tokens)
+  // in the filter_expression matches the size of the parameter sequence.
+  // The tokens start with 0 which means that when the maximum number used is 1 we need
+  // two parameters, (zero and one)
+  if (len != filter_eval_.number_parameters ()) {
```

Comment:
Should a sequence with more elements that required be allowed?  I don't feel strongly about it either way, and the spec doesn't help much.